pkg/store/store.go

```diff
@@ -85,6 +85,45 @@ func (s *MsiKeyVaultStore) GetCredentialsObject(ctx context.Context, secretName
 	return &SecretResponse{CredentialsObject: credentialsObject, Properties: secretProperties}, nil
 }
 
+// Get a deleted credentials object from the key vault using the specified secret name.
+func (s *MsiKeyVaultStore) GetDeletedCredentialsObject(ctx context.Context, secretName string) (*SecretResponse, error) {
+	response, err := s.kvClient.GetDeletedSecret(ctx, secretName, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	if response.Value == nil {
+		return nil, errNilSecretValue
+	}
+
+	var credentialsObject dataplane.CredentialsObject
+	if err := credentialsObject.UnmarshalJSON([]byte(*response.Value)); err != nil {
+		return nil, err
+	}
+
+	secretProperties := SecretProperties{
+		Name:      secretName,
+		Enabled:   true, // Default to true
```

Comment:
If the secret is deleted, it shouldn't be enabled.

Reply:
Also, see attributes section: https://github.com/Azure/azure-sdk-for-go/blob/7664dd24aeda13ec5b7af4531047815c4682ca36/sdk/security/keyvault/azsecrets/models.go#L21

This has Enabled, expires, etc.